app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala

```diff
@@ -78,7 +80,24 @@ class FormProcessor(
     lang: LangADT,
     sse: SmartStringEvaluator
   ): Future[Result] = {
-
+    def computePageLink(
+      formPageId: PageId,
+      pageIdSectionNumberMap: Map[ModelPageId, SectionNumber]
+    ) = {
+      val forModelPageId = formPageId.modelPageId
+      pageIdSectionNumberMap.get(forModelPageId) match {
+        case Some(sectionNumber) => Some(sectionNumber)
+        case None =>
+          pageIdSectionNumberMap.toList
+            .sortBy(_._1.maybeIndex)(Ordering[Option[Int]].reverse)
+            .find { case (modelPageId, _) =>
+              modelPageId.baseId === forModelPageId.baseId
+            }
+            .fold(Option.empty[SectionNumber]) { case (_, sectionNumber) =>
+              Some(sectionNumber)
+            }
```

Comment:
I think all of this should be replaced by
```
def firstSingletonSectionNumber: SectionNumber = iterations.head.singletons.head.sectionNumber
```
which should live next this https://github.com/hmrc/gform-frontend/blob/main/app/uk/gov/hmrc/gform/models/Bracket.scala#L227